### PR TITLE
DISTPG-465 Add ppg-server_builder.sh, rules file for debian builds an…

### PR DESCRIPTION
**Pull request by surbhat1595**
…d update ppg-server.spec file for ppg-server meta-package builds

### CODE REVIEW

File: ppg-server/ppg-server_builder.sh [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+    fi
+    return
+}
+
+add_percona_yum_repo(){
+    if [ ! -f /etc/yum.repos.d/percona-dev.repo ]
+    then
+      wget http://jenkins.percona.com/yum-repo/percona-dev.repo
+      mv -f percona-dev.repo /etc/yum.repos.d/
+    fi
```
Comment (EvgeniyPatlan) [inline ppg-server/ppg-server_builder.sh:81]: this repo is outdated and could be removed

File: ppg-server/ppg-server_builder.sh [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+
+    cd rpmbuild/SRPMS/
+    #
+    cd $WORKDIR
+    RHEL=$(rpm --eval %rhel)
+    ARCH=$(echo $(uname -m) | sed -e 's:i686:i386:g')
+    if [ -f /opt/rh/devtoolset-7/enable ]; then
+        source /opt/rh/devtoolset-7/enable
+        source /opt/rh/llvm-toolset-7/enable
+    fi
```
Comment (EvgeniyPatlan) [inline ppg-server/ppg-server_builder.sh:346]: toolset is not needed as we don't compile anything

File: ppg-server/ppg-server_builder.sh [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+    export LIBPQ_DIR=/usr/pgsql-14/
+    export LIBRARY_PATH=/usr/pgsql-14/lib/:/usr/pgsql-14/include/
+    rpmbuild --define "_topdir ${WORKDIR}/rpmbuild" --define "pginstdir /usr/pgsql-14" --define "dist .$OS_NAME" --define "version ${VERSION}" --rebuild rpmbuild/SRPMS/$SRC_RPM
```
Comment (EvgeniyPatlan) [inline ppg-server/ppg-server_builder.sh:349]: do we need `pginstdir`?

File: ppg-server/ppg-server_builder.sh [hunk 4]
```diff
@@ -0,0 +1,471 @@
...
+          dnf config-manager --set-enabled codeready-builder-for-rhel-8-x86_64-rpms
+          dnf clean all
+          rm -r /var/cache/dnf
+          dnf -y upgrade
+          switch_to_vault_repo
+          yum -y install perl lz4-libs c-ares-devel
+	  yum -y install rpmbuild
+      else
+        yum -y install epel-release
+        yum -y install llvm-toolset-7-clang llvm5.0-devtoolset
```
Comment (EvgeniyPatlan) [inline ppg-server/ppg-server_builder.sh:210]: do we need llvm for metapackage build?

File: ppg-server/ppg-server_builder.sh [hunk 5]
```diff
@@ -0,0 +1,471 @@
...
+      export ARCH=$(echo $(uname -m) | sed -e 's:i686:i386:g')
+      apt-get -y install gnupg2
+      add_percona_apt_repo
+      apt-get update || true
+      INSTALL_LIST="debconf debhelper devscripts dh-exec git wget"
+      until DEBIAN_FRONTEND=noninteractive apt-get -y --allow-unauthenticated install ${INSTALL_LIST}; do
+        sleep 1
+        echo "waiting"
+      done
+      DEBIAN_FRONTEND=noninteractive apt-get -y --allow-unauthenticated install libpam0g-dev || DEBIAN_FRONTEND=noninteractive apt-get -y --allow-unauthenticated install libpam-dev
```
Comment (EvgeniyPatlan) [inline ppg-server/ppg-server_builder.sh:229]: go we need libpam0g-dev ?

**Review by EvgeniyPatlan — APPROVED** [review]

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]
please address comments

**Comment by surbhat1595** [comment]
Hi Evgeniy,

Thanks for the review comments. I have updated the requested changes now.

**Review by EvgeniyPatlan — APPROVED** [review]

File: ppg-server/ppg-server.spec [hunk 6]
```diff
@@ -4,7 +4,7 @@
 
 Summary:        Percona base selection of PostgreSQL%{pgmajorversion} components
 Name:           %{sname}%{pgmajorversion}
-Version:        %{version}
+Version:        %{pgmajorversion}.%{version}
```
Comment (vorsel) [inline ppg-server/ppg-server.spec:7]: Please remove extra two spaces at the end of the line.

**Review by vorsel — APPROVED** [review]

**Pull request merged**